src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -45,10 +48,32 @@ public AddCommand parse(String args) throws ParseException {
                         PREFIX_LICENCE_PLATE, PREFIX_ADDRESS, PREFIX_TAG, PREFIX_POLICY_NUMBER,
                         PREFIX_POLICY_ISSUE_DATE, PREFIX_POLICY_EXPIRY_DATE);
 
+        if (!argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_PREAMBLE_DETECTED));
+        }
+
         if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_NRIC,
-                PREFIX_LICENCE_PLATE, PREFIX_ADDRESS)
-                || !argMultimap.getPreamble().isEmpty()) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+                PREFIX_LICENCE_PLATE, PREFIX_ADDRESS)) {
+            String errorMessage = MESSAGE_MISSING_FIELDS_FOR_ADD_COMMAND;
+            if (argMultimap.getValue(PREFIX_NAME).isEmpty()) {
+                errorMessage += "- Name(" + PREFIX_NAME + ") ";
+            }
+            if (argMultimap.getValue(PREFIX_PHONE).isEmpty()) {
+                errorMessage += "- Phone(" + PREFIX_PHONE + ") ";
+            }
+            if (argMultimap.getValue(PREFIX_EMAIL).isEmpty()) {
+                errorMessage += "- Email(" + PREFIX_EMAIL + ") ";
+            }
+            if (argMultimap.getValue(PREFIX_NRIC).isEmpty()) {
+                errorMessage += "- NRIC(" + PREFIX_NRIC + ") ";
+            }
+            if (argMultimap.getValue(PREFIX_LICENCE_PLATE).isEmpty()) {
+                errorMessage += "- License Plate(" + PREFIX_LICENCE_PLATE + ") ";
+            }
+            if (argMultimap.getValue(PREFIX_ADDRESS).isEmpty()) {
+                errorMessage += "- Address(" + PREFIX_ADDRESS + ") ";
+            }
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, errorMessage));
```

Comment:
Nice way to show the user if any of the fields are missing, it will directly show which are the ones that are missing